src/main/CommandHandler.cpp

```diff
@@ -787,10 +792,33 @@ CommandHandler::generateLoad(std::string const& params, std::string& retStr)
             batchSize = 100;
             retStr = "Setting batch size to its limit of 100.";
         }
-        mApp.generateLoad(isCreate, nAccounts, offset, nTxs, txRate, batchSize);
-        retStr +=
-            fmt::format(" Generating load: {:d} {:s}, {:d} tx/s = {:f} hours",
-                        numItems, itemType, txRate, hours);
+
+        if (spikeSize < txRate)
+        {
+            retStr += "spikesize is less than txrate. No spike will "
+                      "occur.";
+            spikeSize = 0;
+            spikeInterval = std::chrono::seconds(0);
+        }
+
+        if (spikeInterval.count() != 0)
+        {
+            uint32_t numTxsPerInterval =
+                txRate * (spikeInterval.count() - 1) + spikeSize;
+            uint32_t numWholeIntervals = numItems / numTxsPerInterval;
+            hours = spikeInterval.count() * numWholeIntervals;
+            hours += std::min<uint32_t>(
+                spikeInterval.count(),
+                (numItems - numWholeIntervals * numTxsPerInterval) / txRate);
+            hours /= 3600.0;
+        }
+
+        mApp.generateLoad(isCreate, nAccounts, offset, nTxs, txRate, batchSize,
+                          spikeInterval, spikeSize);
+
+        retStr += fmt::format(" Generating load: {:d} {:s}, {:d} tx/s "
```

Comment:
I know this was here before this change, but I'd recommend we remove `hours` ETA altogether -- it's adding complexity and it's not accurate anyway (in perf tests we typically don't generate load for hours, and also current calculation does not account for the batch size during account creation).

Reply:
Done!